packages/stream-model-handler/src/__tests__/model-handler.test.ts

```diff
@@ -993,4 +1050,132 @@ describe('ModelHandler', () => {
 
     expect(state).toMatchSnapshot()
   })
+
+  // it('Anchor Version 1 works correctly', async () => {
+  //   // const genesisCommit = (await Model._makeGenesis(
+  //   //   context.api,
+  //   //   PLACEHOLDER_CONTENT
+  //   // )) as SignedCommitContainer
+  //   // await context.ipfs.dag.put(genesisCommit, FAKE_CID_1)
+
+  //   // const payload = dagCBOR.decode(genesisCommit.linkedBlock)
+  //   // await context.ipfs.dag.put(payload, genesisCommit.jws.link)
+
+  //   // // apply genesis
+  //   // const genesisCommitData = {
+  //   //   cid: FAKE_CID_1,
+  //   //   type: CommitType.GENESIS,
+  //   //   commit: payload,
+  //   //   envelope: genesisCommit.jws,
+  //   // }
+  //   // let state = await handler.applyCommit(genesisCommitData, context)
+
+  //   // const state$ = TestUtils.runningState(state)
+  //   // const doc = new Model(state$, context)
+  //   // const signedCommit = (await doc._makeCommit(
+  //   //   context.api,
+  //   //   FINAL_CONTENT
+  //   // )) as SignedCommitContainer
+
+  //   // await context.ipfs.dag.put(signedCommit, FAKE_CID_2)
+
+  //   // const sPayload = dagCBOR.decode(signedCommit.linkedBlock)
+  //   // await context.ipfs.dag.put(sPayload, signedCommit.jws.link)
+
+  //   // // apply signed
+  //   // const signedCommitData = {
+  //   //   cid: FAKE_CID_2,
+  //   //   type: CommitType.SIGNED,
+  //   //   commit: sPayload,
+  //   //   envelope: signedCommit.jws,
+  //   // }
+  //   // state = await handler.applyCommit(signedCommitData, context, state)
+
+  //   // // apply anchor
+  //   // const anchorProof = {
+  //   //   blockNumber: 123456,
+  //   //   blockTimestamp: 1615799679,
+  //   //   chainId: 'fakechain:123',
+  //   //   // version: 1
+  //   // }
+  //   // await context.ipfs.dag.put(anchorProof, FAKE_CID_3)
+  //   // const anchorCommitData = {
+  //   //   cid: FAKE_CID_4,
+  //   //   type: CommitType.ANCHOR,
+  //   //   commit: { proof: FAKE_CID_3, prev: FAKE_CID_2 },
+  //   //   proof: anchorProof,
+  //   // }
+  //   // state = await handler.applyCommit(anchorCommitData, context, state)
+  //   // delete state.metadata.unique
+  //   // expect(state).toMatchSnapshot()
+  // })
+
+  //could merge into anchor 0 test
+  it('applies version 1 anchor commit correctly', async () => {
```

Comment:
Sorry i'm a bit confused. I don't believe `handler.applyCommit` validates the commit. Please let me know if I am mistaken, or if you are testing something else.

Reply:
yeah I agree these tests aren't really adding a lot of value.  I think a more interesting test would be to update the InMemoryAnchorService to mirror the behavior of the contract-based anchoring system, then the existing tests in the `stream-tests` package that actually anchor real streams against the in-memory anchor service would test the new contract codepaths.  That would be a larger change though, and is still not as valuable as the actual e2e integration tests that use a real anchor service, which we'll be getting anyway and will actually be an easier change.  So I'd maybe file a ticket for updating the InMemoryAnchorService to use smart contract anchoring, but consider it a stretch goal for the project and not really worry about it right now.

The real valuable test coverage we are going to get is from the https://github.com/ceramicnetwork/ceramic-anchor-service/blob/develop/src/__tests__/ceramic_integration.test.ts test in the CAS codebase using Ganache, and from the e2e tests from https://github.com/3box/ceramic-integration-tests that use a real CAS against an Eth testnet

Reply:
> I can knock this out

Let's focus on getting the core changes to js-ceramic and the CAS codebase in first, then we can go back and shore up test coverage second.  The sooner we can get this running in the QA environment the better, then we can add more test coverage as it burns in on QA

Reply:
We could add a new test file for the ethereum validator service because i don't see any tests that are not mocking the validation step. But I agree with spencer, lets focus on getting the core changes because there are dependencies